### DISCUSSION TITLE
Check the number of authors in citekey generation

### DISCUSSION THREAD

**Discussion by anamyk** [retorquere/zotero-better-bibtex#2800] [Q&A]
Hi all,

I like my citekey to be "authorLast + auth + year + "_" + shorttitle(6,6)" as I like to see which group published a work and the lastauthor uses to be the professor which stays more constant.

Now sometimes I import media that has only one author, but the citekey uses the name two times, as authorLast and auth are the same. 

I though I could check for the total number of authors and then use a different formula, e.g.
`auth.items('=',1) auth + year + "_" + shorttitle(6,6) | authorLast + auth + year + "_" + shorttitle(6,6)`

Any idea how to do this or if this is possible? Appreciate any help!

**Comment by retorquere** [comment]
depends on whether you want to literally check for a creator of type `author`, and make sure there is exactly one of those:

```
creators(type=author, min=1, max=1).discard + auth + year + "_" + shorttitle(6,6) | authorLast + auth + year + "_" + shorttitle(6,6)
```

or just whether there is one creator regardless of its type:

```
creators(min=1, max=1).discard + auth + year + "_" + shorttitle(6,6) | authorLast + auth + year + "_" + shorttitle(6,6)
```
